## **2x2 Contigency**
Problem Statement:
Analyze the relationship between teams' wins and losses in cricket matches based on the data provided in the 'matches_won.csv' file. Specifically, determine whether there is a significant association between the teams' performance (wins and losses) in the dataset.

Hypothesis:

Ho: proportions are same

H1: proportions are not same

In [4]:
import pandas as pd

data = pd.read_csv('./data/matches_won.csv')
print(data)



                          Team  Win  lose
0  Royal Challengers Bangalore  121   113
1               Mumbai Indians  140   107


### Perform Chi-Squared Test for Independence

The given data:

- Royal Challengers Bangalore (RCB): Wins = 121, Losses = 113
- Mumbai Indians (MI): Wins = 140, Losses = 107

Let's construct the 2x2 contingency table:

|          | Win   | Lose  | Total |
|----------|-------|-------|-------|
| RCB      | 121   | 113   | 234   |
| MI       | 140   | 107   | 247   |
| Total    | 261   | 220   | 481   |



In [5]:
contingency_table_2x2 = data.iloc[:, 1:].values
print(contingency_table_2x2)


[[121 113]
 [140 107]]


In [12]:

N=contingency_table_2x2[0][0]+contingency_table_2x2[0][1]+contingency_table_2x2[1][0]+contingency_table_2x2[1][1]

#1st method


a=contingency_table_2x2[0][0]
b=contingency_table_2x2[0][1]
c=contingency_table_2x2[1][0]
d=contingency_table_2x2[1][1]



chi_sq_calc=(N*((a*d)-(b*c))**2)/((a+b)*(c+d)*(a+c)*(b+d))
print(chi_sq_calc)


chi_sq_tab=2.706  #for alpha=10% and d.f=1 chi square tabulated=2.706

if chi_sq_calc<=chi_sq_tab:
  print("accept Ho , therefore the proportions are equal")
else:
  print("reject Ho, accept H1 . therefore the proportions are not equal")


1.1963006247186525
[[126.97297297297297, 107.02702702702703], [134.02702702702703, 112.97297297297297]]
accept Ho , therefore the proportions are equal


In [15]:
#2nd method

expected_values = [[0, 0], [0, 0]]

for i in range(2):  # Iterate through rows
    row_sum = sum(contingency_table_2x2[i])  # Calculate row sum
    for j in range(2):  # Iterate through columns
        col_sum = sum(row[j] for row in contingency_table_2x2)  # Calculate column sum
        expected_values[i][j] = (row_sum * col_sum) / N  # Compute expected value

print(expected_values)
chi_cal=0
for i in range(2):
   for j in range(2):
      chi_cal=chi_cal+((contingency_table_2x2[i][j]-expected_values[i][j])**2/expected_values[i][j])


print(chi_cal)


chi_sq_tab=2.706  #df=1 alpha=0.1

if chi_cal<=chi_sq_tab:
  print("accept Ho , therefore the proportions are equal")
else:
  print("reject Ho, accept H1 . therefore the proportions are not equal")


[[126.97297297297297, 107.02702702702703], [134.02702702702703, 112.97297297297297]]
1.1963006247186507
accept Ho , therefore the proportions are equal
